Java migration: language edit of 4 to 5 section (#2697)
cap-js/docs: java/migration.md

| [cds.security.authentication.mode](./developing-applications/properties#cds-security-authentication) | `model-strict` | `model-relaxed` | Authentication mode now defaults to `model-relaxed`, which only enforces authentication for endpoints protected via `@requires` or `@restrict`. |
| [cds.sql.hana.search.fuzzy](./developing-applications/properties#cds-sql-hana-search-fuzzy) | `false` | `true` | Fuzzy search is now enabled on HANA by default. |
| [cds.sql.toOnePath.mode](./developing-applications/properties#cds-sql-toonepath-mode) | `always-join` | `optimize` | SQL generation now avoids joins for to-one path expressions when a FK column can be selected directly, improving query performance. |

### Deprecated Properties
<div id="deprecated-properties-4-to-5" />

The following properties are deprecated and may be removed in a future major version:

| Deprecated Property | Explanation |
| --- | --- |
| `cds.dashboard.*` | The entire `cds.dashboard` configuration namespace is deprecated and may be removed in a future major version. |
| `cds.outbox.inMemory.emitDuringChangeSetContext` | This functionality is always enabled by default. You cannot switch it off. |

> | `cds.outbox.inMemory.enabled` | This functionality is always enabled by default. You cannot switch it off. |

Reviewer: ```suggestion
| `cds.outbox.inMemory.emitDuringChangeSetContext` | This functionality is now always enabled. |
| `cds.outbox.inMemory.enabled` | This functionality is now always enabled. |
```

[Learn more about Maven setup with Lombok.](https://projectlombok.org/setup/maven){.learn-more}
[Learn more about the change in the Java compiler.](https://bugs.java.com/bugdatabase/JDK-8321314/description){.learn-more}

### Removed Repackaged Olingo Dependencies { #removed-olingo-4-to-5 }

The internal Maven modules `repackaged/odata-v4-lib` and `repackaged/odata-v2-lib` have been removed. If your project directly references these modules and does not compile after migrating to CAP Java 5.x, you have three options to maintain compatibility:

1. [**Recommended**] Remove `mvn` dependencies on `repackaged/odata-v4-lib` and `repackaged/odata-v2-lib`, and rewrite code using CAP Java native APIs.

2. Use `maven-dependency-plugin` to unpack (extract contents of dependency) Olingo sources into your codebase during the project build.

3. Use upstream open-source Apache Olingo. Change dependencies from the corresponding internal CAP `mvn` modules to OSS packages `org.apache.olingo:olingo-odata4` and `org.apache.olingo:olingo-odata2`.

> #### Module `com.sap.cds:cds4j-codegen` Is Removed

Reviewer: ```suggestion
#### Module `com.sap.cds:cds4j-codegen` is Removed
```
